src/api/controllers/ItemImageController.ts

```diff
@@ -29,14 +37,88 @@ export class ItemImageController {
         return res.found(itemImages.toJSON());
     }
 
-    @httpPost('/')
-    public async create( @response() res: myExpress.Response, @requestBody() body: any, @request() req: any): Promise<any> {
+    @httpPost('/template/:templateId')
+    public async create( @response() res: myExpress.Response, @requestParam('templateId') templateIdStr: string,
+                         @requestBody() body: any, @request() req: any): Promise<any> {
+        // check templateId is number
+        // also check listingItemTemplate id present in params
+        let templateId: number;
+        if (!templateIdStr) {
+            throw new MessageException('ListingItemTemplate id can not be null.');
+        } else {
+            templateId = parseInt(templateIdStr, 10);
+            if ( !templateId ) {
+                throw new MessageException('ListingItemTemplate id must be an integer.');
+            }
+        }
+
+        // this.log.debug('templateId: ' + templateId);
+        // this.log.debug('files: ', req.files);
+
+        for ( const file of req.files ) {
+            // this.log.debug(`Found image <${file.fieldname}>.`);
+            if ( file.fieldname === 'image' ) {
+                const imageFile = file;
+
+                // Read the file data in
+                const dataStr = fs.readFileSync(imageFile.path, 'base64');
+                // this.log.error('dataStr = ' + dataStr);
+
+                // find listing item template
+                const listingItemTemplate = await this.listingItemTemplateService.findOne(templateId);
+                this.log.debug('imageFile.mimetype = ' + imageFile.mimetype);
```

Comment:
listingItemTemplate will be the same for all images, but you are fetching it over and over again in a loop.

Reply:
This is true, but we only expect one file to be uploaded with the fieldname 'image', so shouldn't happen multiple times